test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATRemovePrefix to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATRemovePrefix_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
+
+    /* Check that CELLULAR_AT_SUCCESS is returned . */
+    cellularStatus = Cellular_ATRemovePrefix( &pString );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+
+    /* Test the result to check its functionality. */
+    TEST_ASSERT_EQUAL_STRING( CELLULAR_SAMPLE_PREFIX_STRING_OUTPUT, pString );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATRemovePrefix to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATRemovePrefix_Error_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = "";
+    char * pWrongString = CELLULAR_SAMPLE_PREFIX_STRING_WRONG_INPUT;
+    char * pLargeString = CELLULAR_SAMPLE_PREFIX_STRING_LARGE_INPUT;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATRemovePrefix( &pString );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATRemovePrefix( &pWrongString );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATRemovePrefix( &pLargeString );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATRemoveLeadingWhiteSpaces to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATRemoveLeadingWhiteSpaces_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATRemoveLeadingWhiteSpaces( NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATRemoveLeadingWhiteSpaces to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATRemoveLeadingWhiteSpaces_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = CELLULAR_SAMPLE_STRING_LEADING_WHITE_SPACE;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATRemoveLeadingWhiteSpaces( &pString );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+
+    /* Test the result to check its functionality. */
+    TEST_ASSERT_EQUAL_STRING( CELLULAR_SAMPLE_STRING_NO_WHITE_SPACE, pString );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATRemoveLeadingWhiteSpaces to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATRemoveLeadingWhiteSpaces_Error_Path( void )

```

Comment:
Error_Path seems to generic. How about Empty_String instead?

Reply:
sounds good!